e2e-test/src/test/java/org/alfresco/hxi_connector/e2e_test/QuestionsPermissionsE2eTest.java

```diff
@@ -0,0 +1,178 @@
+import org.alfresco.hxi_connector.e2e_test.util.client.RepositoryClient;
+import org.alfresco.hxi_connector.e2e_test.util.client.model.User;
+
+@Testcontainers
+@SuppressWarnings("PMD.FieldNamingConventions")
+public class QuestionsPermissionsE2eTest
+{
+    private static final String QUESTIONS_URL = "/alfresco/api/-default-/private/hxi/versions/1/questions";
+    private static final String SAMPLE_QUESTION = """
+            {
+                "question": "What is the meaning of life?",
+                "agentId": "agent-id",
+                "restrictionQuery": {
+                    "nodesIds": ["%s"]
+                }
+            }
+            """;
+    static final Network network = Network.newNetwork();
+    @Container
+    static final PostgreSQLContainer<?> postgres = DockerContainers.createPostgresContainerWithin(network);
+    @Container
+    static final GenericContainer<?> activemq = DockerContainers.createActiveMqContainerWithin(network);
+    @Container
+    private static final WireMockContainer hxInsightMock = DockerContainers.createWireMockContainerWithin(network)
+            .withFileSystemBind("src/test/resources/wiremock/hxinsight", "/home/wiremock", BindMode.READ_ONLY);
+    @Container
+    static final AlfrescoRepositoryContainer repository = createRepositoryContainer()
+            .dependsOn(postgres, activemq);
+    private static final User regularUser = new User("test", "test");
+    private static String publicDocumentId;
+    private static String privateDocumentId;
+
+    @BeforeAll
+    public static void beforeAll()
+    {
+        RepositoryClient repositoryClient = new RepositoryClient(repository.getBaseUrl(), ADMIN_USER);
+        WireMock.configureFor(hxInsightMock.getHost(), hxInsightMock.getPort());
+
+        repositoryClient.createUser(regularUser);
+
+        String publicSiteId = repositoryClient.createSite("Test Public Project", PUBLIC);
+        String publicSiteRootDirectory = repositoryClient.getSiteDocumentLibraryId(publicSiteId);
+        publicDocumentId = repositoryClient.createNodeWithContent(
+                publicSiteRootDirectory,
+                "test public file",
+                new ByteArrayInputStream("test file content".getBytes()),
+                "text/plain").id();
+
+        String privateSiteId = repositoryClient.createSite("Test Private Project", PRIVATE);
+        String privateSiteRootDirectory = repositoryClient.getSiteDocumentLibraryId(privateSiteId);
+        privateDocumentId = repositoryClient.createNodeWithContent(
+                privateSiteRootDirectory,
+                "test private file",
+                new ByteArrayInputStream("test file content".getBytes()),
+                "text/plain").id();
+    }
+
+    @Test
+    void adminShouldBeAbleToAskQuestionAboutPublicDocument()
+    {
+        // when
+        Response response = given().auth().preemptive().basic(ADMIN_USER.username(), ADMIN_USER.password())
```

Comment:
Maybe it's worth considering creating in `RepositoryClient` methods like `askQuestion(question)` and `askQuestionAs(question, user)` and move the code out from tests - WDYT?

Reply:
I don't like it tbh, I would prefer to stay as explicit as possible even if this means some repetitions

Reply:
I'm fine with that